examples/getting-started/src/main/java/com/google/cloud/sql/mysql/example/GettingStarted.java

```diff
@@ -0,0 +1,327 @@
+      return;
+    }
+
+    Optional<String> database = askForDatabase(databases);
+    if (!database.isPresent()) {
+      return;
+    }
+
+    printConnectionDetails(instanceConnectionName, database, optionalMysqlCredentials.get());
+  }
+
+  private List<String> listDatabases(Connection connection) throws SQLException {
+    Statement statement = connection.createStatement();
+    ResultSet resultSet = statement.executeQuery("SHOW DATABASES");
+    List<String> databases = new ArrayList<>();
+    while (resultSet.next()) {
+      String database = resultSet.getString("database");
+      if (SYSTEM_DATABASES.contains(database)) {
+        continue;
+      }
+      databases.add(database);
+    }
+    statement.close();
+    databases.sort(String::compareTo);
+    return databases;
+  }
+
+  private Optional<MysqlCredentials> askForMysqlCredentials(String instanceConnectionName)
+      throws SQLException {
+    Console console = System.console();
+    String user;
+    String lastUser = "root";
+    for (; ; ) {
+      char[] password;
+      System.out.printf("Please enter MySQL username [%s]: ", lastUser);
+      user = console.readLine();
+      if (user == null) {
+        return Optional.empty();
+      }
+
+      if (user.trim().isEmpty()) {
+        user = lastUser;
+      } else {
+        lastUser = user;
+      }
+
+      System.out.print("Please enter MySQL password: ");
+      password = console.readPassword();
+      if (password == null) {
+        return Optional.empty();
+      }
+
+      try {
+        return Optional.of(
+            new MysqlCredentials(
+                user,
+                password,
+                DriverManager.getConnection(
+                    constructJdbcUrl(instanceConnectionName, "mysql"),
+                    user,
+                    new String(password))));
+      } catch (SQLException e) {
+        if (e.getErrorCode() == 1045) {
+          System.out.println("Invalid username/password. Please try again.");
+          continue;
+        }
+        throw e;
+      }
+    }
+  }
+
+  private static String constructJdbcUrl(String instanceConnectionName, String database) {
+    return String.format(
+        "jdbc:mysql://google/%s?cloudSqlInstance=%s&"
+            + "socketFactory=com.google.cloud.sql.mysql.SocketFactory",
+        database,
+        instanceConnectionName);
+  }
+
+  private Optional<String> askForInstance(List<DatabaseInstance> instances) {
+    Console console = System.console();
+
+    for (int i = 0; i < instances.size(); i++) {
+      DatabaseInstance databaseInstance = instances.get(i);
+      System.out.println(String.format("%d: %s (%s)", i + 1, databaseInstance.getName(),
+          databaseInstance.getConnectionName()));
+    }
+
+    String instanceConnectionName = "";
+    while (instanceConnectionName.isEmpty()) {
+      System.out.print("Please enter the number of the instance you want to use [1]: ");
+      String line = console.readLine();
```

Comment:
Might be worth extracting the "select from a list" logic into its own method. It'd only have two users though.